ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -83,6 +85,27 @@ public synchronized void startForMaintenance(ServiceProvider<? super Maintainabl
   private void internalStart() {
     if (!started) {
       createLocationIfRequiredAndVerify(rootDirectory);
+      if (!clean.exists()) {
+        if (lockFile.exists()) {
+          onStartupIfLockFileExists();
+        } else {
+          LOGGER.debug("lock file is not exists.");
+        }
+        if (tryRecursiveDelete(rootDirectory)) {
```

Comment:
Sorry, I hinted to this in my previous review but probably wasn't explicit enough. The `LocalPersistenceService` implementation cannot be this dogmatic about what the unclean state of a directory means to the consumers of the service. It happens that all the existing OSS users of this service don't tolerate an unclean shutdown. That's not true for all the existing users of this service, and isn't necessarily true for all future OSS users. We'll need to find a way that the `LocalPersistenceService` can communicate the clean/unclean state of the service to consumers so they can make a decision on a case by case basis as to whether their files need to be deleted.